### DISCUSSION TITLE
Need Chapter level meta-Data

### DISCUSSION THREAD

**Discussion by BimleshTrigyn** [Sunbird-Ed/Community#494] [Feature request]
Hi Sunbird team,

For any content ID's(Do_Id's) , 
How to get the chapter level met-data..?
I've checked , there is no any API to get the chapter level met-data., could you please implement it as a feature request, where if we are passing any content ID's(Do_Id's) then , it will return the chapter level meta-data of the content ID's(Do_Id's).

Thanks.

**Comment by chitranshu-keshav** [comment]
@rajeevsathish Can you please get back on the above query

**Comment by rajeevsathish** [comment]
@BimleshTrigyn there is a API to get the chapter level for a collection id we can use the hierarchy api call 
for example for a collection with do ID do_312774547315761152110267 the chapter can be seen from children object
please refer this API for more details
v1/hierarchy/do_312774547315761152110267
cc: @Krishnaj20 @chitranshu-keshav @vinukumar-vs

**Reply by BimleshTrigyn** [reply]
Hii @rajeevsathish ,

Thanks for the response, can you plz provide me the API curl or link where will get all the required details(header , body...) for the  above mentioned API.

Thanks.

**Reply by Krishnaj20** [reply]
@BimleshTrigyn - content Do ids will not have any child nodes (chapters..). Are you referring to collection do ids and the corresponding child node (chapters ..) meta data?

@kameshbhr

**Comment by BimleshTrigyn** [comment]
Hii @rajeevsathish ,

Yes I'm trying to get the chapter details of the content id's(corresponding child node)., 
if there is any API for collection also, to get the chapter level meta-data please  share the link.

If not, then  can you please implement it as a feature request.

Thanks.

**Reply by Krishnaj20** [reply]
Hi @BimleshTrigyn - Looks like this request is similar to https://github.com/orgs/Sunbird-Ed/discussions/469 where a response was provided. Can you clarify?

@kameshbhr

**Reply by BimleshTrigyn** [reply]
Hii @Krishnaj20 ,

Yes,  I was discuss earlier.
But the above request (https://github.com/orgs/Sunbird-Ed/discussions/469) , has not fulfilled the user requirement,

User wants the chapter level meta-data of any content-id's do_id's ,independently (no parent and child hierarchy involvement).

Thanks.

**Comment by rajeevsathish** [comment]
@BimleshTrigyn a content cannot have child noted so only collection can have child node that can be got using the hierarchy API
please refer this api this a GET call `https://diksha.gov.in/api/course/v1/hierarchy/do_3131034750542643201961`
and for a content
it will be only content read api call `https://diksha.gov.in/api/content/v1/read/do_31323725777331814413694`

**Reply by BimleshTrigyn** [reply]
Hii @rajeevsathish  , 
for content read API call https://diksha.gov.in/api/content/v1/read/do_31323725777331814413694 ,there is no meta-data like chapter in the response.
please see the attach API response.

{
    "id": "api.content.read",
    "ver": "1.0",
    "ts": "2023-05-23T11:23:41.330Z",
    "params": {
        "resmsgid": "454c7720-f95c-11ed-999c-e79edbde7d8c",
        "msgid": "454bdae0-f95c-11ed-a9a1-b1a28ad966f9",
        "status": "successful",
        "err": null,
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "content": {
            "ownershipType": [
                "createdBy"
            ],
            "unitIdentifiers": [
                "do_4313133211381104640130646"
            ],
            "copyright": "Sri Sathya Sai Central Trust",
            "organisationId": "0c8554e6-95ab-40e6-84f0-647c85f73c07",
            "previewUrl": "https://obj.diksha.gov.in/ntp-content-production/content/html/do_31323725777331814413694-latest",
            "keywords": [
                ""
            ],
            "subject": [
                "Environmental Studies"
            ],
            "channel": "0125196274181898243",
            "downloadUrl": "https://obj.diksha.gov.in/ntp-content-production/content/do_31323725777331814413694/chhotus-house_sa_floral-rangoli_1679294381444_do_31323725777331814413694_2.ecar",
            "language": [
                "English"
            ],
            "source": "https://vdn.diksha.gov.in/api/content/v1/read/do_43132358303086837761658",
            "mimeType": "application/vnd.ekstep.html-archive",
            "variants": {
                "full": {
                    "ecarUrl": "https://obj.diksha.gov.in/ntp-content-production/content/do_31323725777331814413694/chhotus-house_sa_floral-rangoli_1679294381444_do_31323725777331814413694_2.ecar",
                    "size": "1484983"
                },
                "spine": {
                    "ecarUrl": "https://obj.diksha.gov.in/ntp-content-production/content/do_31323725777331814413694/chhotus-house_sa_floral-rangoli_1679294382525_do_31323725777331814413694_2_SPINE.ecar",
                    "size": "37459"
                }
            },
            "objectType": "Content",
            "se_mediums": [
                "English"
            ],
            "gradeLevel": [
                "Class 3"
            ],
            "appIcon": "https://obj.diksha.gov.in/ntp-content-production/content/do_31323725777331814413694/artifact/sssvv.thumb.jpg",
            "primaryCategory": "Teacher Resource",
            "appId": "prod.diksha.portal",
            "contentEncoding": "gzip",
            "artifactUrl": "https://obj.diksha.gov.in/ntp-content-production/content/do_31323725777331814413694/artifact/chhotu_s-house_sa_floral-rangoli.zip",
            "me_totalPlaySessionCount": {
                "portal": 2
            },
            "sYS_INTERNAL_LAST_UPDATED_ON": "2023-03-25T21:08:52.848+0000",
            "contentType": "MarkingSchemeRubric",
            "se_gradeLevels": [
                "Class 3"
            ],
            "identifier": "do_31323725777331814413694",
            "audience": [
                "Teacher"
            ],
            "me_totalTimeSpentInSec": {
                "portal": 28
            },
            "visibility": "Default",
            "consumerId": "19004ef4-9851-4850-89b2-8f6ddca5103c",
            "mediaType": "content",
            "osId": "org.ekstep.quiz.app",
            "languageCode": [
                "en"
            ],
            "lastPublishedBy": "a31407f7-7682-4e7d-b191-ba1442dd73ad",
            "version": 2,
            "se_subjects": [
                "Environmental Studies"
            ],
            "license": "CC BY 4.0",
            "prevState": "Review",
            "size": 1484867,
            "lastPublishedOn": "2023-03-20T06:39:38.442+0000",
            "name": "Chhotu's House_SA_Floral Rangoli",
            "attributions": [
                "Sri Sathya Sai Vidya Vahini"
            ],
            "status": "Live",
            "code": "799513ab-0446-3b6f-e512-2c7a9c482c8a",
            "credentials": {
                "enabled": "No"
            },
            "prevStatus": "Processing",
            "origin": "do_43132358303086837761658",
            "description": "",
            "streamingUrl": "https://obj.diksha.gov.in/ntp-content-production/content/html/do_31323725777331814413694-latest",
            "medium": [
                "English"
            ],
            "idealScreenSize": "normal",
            "createdOn": "2021-03-16T06:29:53.033+0000",
            "processId": "6e19357f-1bc9-4d93-a674-a0088be3a2cc",
            "contentDisposition": "inline",
            "lastUpdatedOn": "2023-03-20T06:39:42.611+0000",
            "originData": {
                "identifier": "do_43132358303086837761658",
                "repository": "https://vdn.diksha.gov.in/api/content/v1/read/do_43132358303086837761658"
            },
            "collectionId": "do_4313133211355660288129998",
            "dialcodeRequired": "No",
            "lastStatusChangedOn": "2023-03-20T06:39:42.611+0000",
            "createdFor": [
                "0125196274181898243"
            ],
            "creator": "Sri Sathya Sai Vidya Vahini",
            "os": [
                "All"
            ],
            "se_FWIds": [
                "ncert_k-12"
            ],
            "pkgVersion": 2,
            "versionKey": "1615876197300",
            "migrationVersion": 1.1,
            "idealScreenDensity": "hdpi",
            "framework": "ncert_k-12",
            "s3Key": "ecar_files/do_31323725777331814413694/chhotus-house_sa_floral-rangoli_1615876197446_do_31323725777331814413694_1.0.ecar",
            "lastSubmittedOn": "2021-03-16T06:29:56.198+0000",
            "createdBy": "b97638a3-9376-4731-bb71-55e0181bc021",
            "compatibilityLevel": 1,
            "programId": "d4259b30-129a-11eb-9229-4bcf1c6ae295"
        }
    }
}

I'm asking to implement it in the content read API , so that the content read API will return chapter meta-data, along with existing meta-data.

Thanks.

**Reply by kameshbhr** [reply]
@BimleshTrigyn, looks like there is a confusion in your understanding regarding what a chapter is. 
A chapter is **not** metadata of a content like Class, Medium etc. It is a folder in a Digital Textbook. A Digital Textbook is a collection. So a content as is **cannot** have metadata such as a Chapter. A content can be "added" to a Chapter in a Digital Textbook. It means the content is linked to a folder in a collection. So the information of "content belongs to a chapter" only makes sense when it is linked to a folder (chapter) within a collection (digital textbook).
I think it would be better to get into a quick call to get clarity on this along with a product manager and architect from your team.

**Reply by BimleshTrigyn** [reply]
@kameshbhr , Now got it,
 Yes I was having some confusion , I was thinking that ,we can implement it in the content read API.

Thanks.

**Reply by kameshbhr** [reply]
No problem. Feel free to ask any queries that you may have. We can also plan sessions if you need more clarity on any specific topics.

**Reply by Krishnaj20** [reply]
@BimleshTrigyn - locking this thread as your query is answered.
